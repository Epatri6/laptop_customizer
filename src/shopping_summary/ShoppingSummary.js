import React from 'react';
import ShoppingItem from '../shopping_item/ShoppingItem';

export default class ShoppingSummary extends React.Component {

  //Prints the items selected
  summary = () => {
    const selectedFeatures = this.props.selectedFeatures;
    return Object.keys(selectedFeatures).map((feature, idx) => {
      const featureHash = feature + '-' + idx;
      const selectedOption = selectedFeatures[feature];
      return (
        <ShoppingItem
          key={featureHash}
          id={featureHash}
          feature={feature}
          selectedOption={selectedOption}
        />
      );
    });
  };

  render() {
    return (
      <div>
        {this.summary()}
      </div>
    );
  }

  static defaultProps = {
    selectedFeatures: {}
  }
}
