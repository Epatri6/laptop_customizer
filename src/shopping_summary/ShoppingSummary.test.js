import React from 'react';
import ReactDOM from 'react-dom';
import ShoppingSummary from './ShoppingSummary';
import renderer from 'react-test-renderer';

describe('Testing ShoppingSummary Component', () => {

  const testProps = {
    Processor: {
      name: '17th Generation Intel Core HB (7 Core with donut spare)',
      cost: 700
    },
    'Operating System': {
      name: 'Ubuntu Linux 16.04',
      cost: 200
    }
  };

  //Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingSummary selectedFeatures={testProps}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders without props without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ShoppingSummary />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //Snapshot test
  it('renders as expected', () => {
    const tree = renderer
      .create(<ShoppingSummary selectedFeatures={testProps}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });
  
  it('renders as expected without props', () => {
    const tree = renderer
      .create(<ShoppingSummary />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});