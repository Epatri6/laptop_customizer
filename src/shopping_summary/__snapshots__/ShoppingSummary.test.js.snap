// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ShoppingSummary Component renders as expected 1`] = `
<div>
  <div
    className="summary__option"
  >
    <div
      className="summary__option__label"
    >
      Processor
       
    </div>
    <div
      className="summary__option__value"
    >
      17th Generation Intel Core HB (7 Core with donut spare)
    </div>
    <div
      className="summary__option__cost"
    >
      $700.00
    </div>
  </div>
  <div
    className="summary__option"
  >
    <div
      className="summary__option__label"
    >
      Operating System
       
    </div>
    <div
      className="summary__option__value"
    >
      Ubuntu Linux 16.04
    </div>
    <div
      className="summary__option__cost"
    >
      $200.00
    </div>
  </div>
</div>
`;

exports[`Testing ShoppingSummary Component renders as expected without props 1`] = `<div />`;
